presto-mongodb/src/main/java/io/prestosql/plugin/mongodb/MongoSession.java

```diff
@@ -495,6 +497,10 @@ private boolean deleteTableMetadata(SchemaTableName schemaTableName)
 
         MongoDatabase db = client.getDatabase(schemaName);
         if (!collectionExists(db, tableName)) {
+            if (!db.getCollection(META_TABLE_NAME).find(Filters.eq(TABLE_NAME_KEY, tableName)).first().isEmpty()) {
+                DeleteResult result = db.getCollection(META_TABLE_NAME).deleteOne(Filters.eq(TABLE_NAME_KEY, tableName));
+                return result.getDeletedCount() == 1;
+            }
```

Comment:
How about the following to be consistent?
```java
        if (!collectionExists(db, tableName) &&
                db.getCollection(schemaCollection).find(new Document(TABLE_NAME_KEY, tableName)).first().isEmpty()) {
            return false;
        }
```

Reply:
Thanks, your suggestion is better